src/xml/builder.cr

```diff
@@ -44,12 +44,19 @@ struct XML::Builder
 
   # Emits the start of an element.
   def start_element(name : String) : Nil
-    call StartElement, string_to_unsafe(name)
+    check_valid_element_name name, string_to_unsafe(name), "element name"
+    call StartElement, name
```

Comment:
Isn't this changing the argument?

Shouldn't it be:

```cr
    unsafe_name = string_to_unsafe(name)
    check_valid_element_name name, unsafe_name, "element name"
    call StartElement, unsafe_name
```

Reply:
Oh, good catch.  Wonder why that didnt cause things to brek..

Reply:
It's the same

Reply:
Actually it's better the new way to avoid double call